cert.go

```diff
@@ -257,6 +257,25 @@ func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *t
 		}
 
 		apisMu.RLock()
+
```

Comment:
Since this code handles client certificates for mutual tls and server certificates - Can we add comments for it. Something along these lines:
261: // Dynamically add API specific certificates for the server authentication
278: // Add certificate for Client authentication for the relevant domain
250: // Add certificate for Client authentication for ControlAPI